### ISSUE TITLE
Undoing the Create camera entity from view action crashes the Editor

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#6655]
**Describe the bug**
Undoing the creation of camera entity from view crashes the Editor.

Please refer to the attached video.

**Steps to reproduce**
1. Launch any level.
2. Right click the Viewport and select _Create camera entity from view_.
3. Press _CTRL+Z_.

**Expected behavior**
The action is undone. The Editor remains stable.

**Actual behavior**
The Editor crashes.

**Call Stack Trace**
```
 7) 00007FFCC296B7F7 (KERNELBASE) : UnhandledExceptionFilter
 6) 00007FFCC4CF51B0 (ntdll) : memset
 5) 00007FFCC4CDC766 (ntdll) : _C_specific_handler
 4) 00007FFCC4CF20CF (ntdll) : _chkstk
 3) 00007FFCC4CA1454 (ntdll) : RtlRaiseException
 2) 00007FFCC4CF0BFE (ntdll) : KiUserExceptionDispatcher
 1) 0000000000010100 (module-name not available) : (function-name not available)
```

**Assets required**
N/A

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/148063673-b12b9f11-4491-4510-88b7-b1e5b0f31b67.mp4

**Found in Branch**
Development (7581967)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

**Attachments**
[Editor.log](https://github.com/o3de/o3de/files/7807765/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7807766/error.log)
[error.zip](https://github.com/o3de/o3de/files/7807767/error.zip)

**Comment by hultonha** [comment]
Is this something you can take a look at please @AMZN-Igarri? We can discuss tomorrow but should hopefully be easy to track down (seems like a solid repro). Cheers!

**Comment by AMZN-Igarri** [comment]
This was fixed with #6785

**Comment by LB-TomaszSerwanski** [comment]
Issue verified as fixed in Development (f5fcab7).
